common/membership/ringpop/service_resolver.go

```diff
@@ -249,50 +257,41 @@ func (r *serviceResolver) HandleEvent(
 		// Note that we receive events asynchronously, possibly out of order.
 		// We cannot rely on the content of the event, rather we load everything
 		// from ringpop when we get a notification that something changed.
-		if err := r.refresh(); err != nil {
+		if err := r.refresh(true); err != nil {
 			r.logger.Error("error refreshing ring when receiving a ring changed event", tag.Error(err))
 		}
 	}
 }
 
-func (r *serviceResolver) refresh() error {
+func (r *serviceResolver) refresh(force bool) error {
```

Comment:
IMHO having two functions `refresh` and `forceRefresh` makes call sites easier to read then `refresh(true)` and `refresh(false)`. Your thoughts?